lib/index.d.ts

```diff
@@ -295,3 +361,31 @@ export class Component<
    */
   getContext<ContextKey extends keyof ContextRegistryT>(contextName: ContextKey): ContextRegistryT[ContextKey];
 }
+
+/**
+ * Panel component that only accepts 3 generic types
+ */
+export class ParamComponent<ParamT = unknown, StateT = unknown, ContextRegistryT = unknown> extends Component<
+  StateT,
+  unknown,
+  unknown,
+  unknown,
+  ContextRegistryT,
+  ParamT
+> {}
+
+// define jsx IntrinsicElement inside namespace jsx to play well with react
+declare global {
```

Comment:
this is a bit interesting, i found out by installing react and try to make snabbdom jsx and react jsx to play well together. after some trial and errors, it seems like the only way to do it is

1. put this global namespace in panel
2. since our jsxPragma is `jsx`, we need to make a namespace with `JSX.IntrinsicElements` inside it for typescript to correctly infer we are on snabbdom jsx.